src/vsr/replica.zig

```diff
@@ -2208,14 +2606,24 @@ pub fn Replica(
                         assert(m.header.cluster == self.cluster);
                         assert(m.header.replica == replica);
                         assert(m.header.view == self.view);
+                        assert(m.header.commit <= self.commit_max);
 
                         if (replica != self.replica) {
-                            if (m.header.op < op) nacks += 1;
-
-                            log.debug("{}: discard_uncommitted_headers: replica={} op={}", .{
+                            // Check for a gap in the uncommitted headers from this replica.
+                            var nack: bool = true;
+                            for (self.message_body_as_headers(m)) |*h| {
+                                if (h.op == op) {
+                                    nack = false;
+                                    break;
+                                }
+                            }
```

Comment:
Could we perhaps have a correctness bug here, counting a nack when we shouldn't, related to how the `do_view_change` message contains the latest N prepare headers of the log suffix from a replica?

For example, if replicas include their latest 3 headers in the DVC, and if replica A is ahead, and has prepares 7,8,9 in its DVC and other replicas have 4,5,6 in their DVC then could this not mean that 4,5,6 get an extra nack count from replica A where they shouldn't?

It seems we should be careful here to count only within the range of what a replica actually sent, without counting anything that has to be before their latest N headers.